docs/cvl/types.md

```diff
@@ -308,3 +297,39 @@ contract function call.
 This section is incomplete.  See [the old documentation](/docs/confluence/anatomy/ghostfunctions).
 ```
 
+(type-conversions)=
+Conversions between CVL and Solidity types
+------------------------------------------
+
+When a specification calls a contract function, the Prover must convert the
+arguments from their CVL types to the corresponding Solidity types, and must
+convert the return values from Solidity back to CVL.  The Prover must also apply
+these conversions when inlining {ref}`hooks <hooks>` and {ref}`function
+summaries <function-summary>`.
+
+There are restrictions on what types can be converted from CVL to Solidity and
+vice-versa.  In general, if a contract uses a type that is not convertible, you
+can still interact with it, but you cannot access the corresponding values. For
+example, if a contract function returns a type that isn't convertible to CVL,
+you can call the function from CVL but you cannot access its return type.
+Similarly, if the function accepts an argument of a type that is not
```

Comment:
So this is one thing @jtoman's work solves maybe with some caveats or exceptions that should be documented